src/mlpack/tests/main_tests/range_search_test.cpp

```diff
@@ -0,0 +1,389 @@
+    if (!data::Load("iris_test.csv", querydata))
+      BOOST_FAIL("Unable to load dataset iris_test.csv!");
+
+    SetInputParam("min", minv);
+    SetInputParam("max", maxv);
+    SetInputParam("distances_file", distancefile);
+    SetInputParam("neighbors_file", neighborsfile);
+    SetInputParam("query", querydata);
+    SetInputParam("reference", inputdata);
+    SetInputParam("tree_type", trees[i]);
+
+    mlpackMain();
+
+    neighborstemp = ReadData<size_t>(neighborsfile);
+    distancestemp = ReadData<double>(distancefile);
+
+    CheckMatrices(neighbors, neighborstemp);
+    CheckMatrices(distances, distancestemp);
+    BOOST_REQUIRE_EQUAL(CheckModelSerial(outputmodel1,
+                                   CLI::GetParam<RSModel*>("output_model")), 0);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(RandomBasisTesting)
+{
+  string distancefile = "distances.csv";
+  string neighborsfile = "neighbors.csv";
+  double minv = 0, maxv = 3;
+  arma::mat querydata, inputdata;
+  vector<vector<size_t>> neighbors, neighborstemp;
+  vector<vector<double>> distances, distancestemp;
+  if (!data::Load("iris.csv", inputdata))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+  if (!data::Load("iris_test.csv", querydata))
+    BOOST_FAIL("Unable to load dataset iris_test.csv!");
+
+  SetInputParam("min", minv);
+  SetInputParam("max", maxv);
+  SetInputParam("distances_file", distancefile);
+  SetInputParam("neighbors_file", neighborsfile);
+  SetInputParam("reference", inputdata);
+
+  mlpackMain();
+
+  RSModel* outputmodel = move(CLI::GetParam<RSModel*>("output_model"));
+
+  SetInputParam("min", minv);
+  SetInputParam("max", maxv);
+  SetInputParam("distances_file", distancefile);
+  SetInputParam("neighbors_file", neighborsfile);
+  SetInputParam("reference", inputdata);
+  SetInputParam("random_basis",true);
+
+  mlpackMain();
+
+  BOOST_REQUIRE_EQUAL(CheckModelSerial(outputmodel,
+                                    CLI::GetParam<RSModel*>("output_model")),0);
+}
+BOOST_AUTO_TEST_CASE(NaiveModeTest)
+{
+  string distancefile = "distances.csv";
+  string neighborsfile = "neighbors.csv";
+  double minv = 0, maxv = 3;
+  arma::mat querydata, inputdata;
+  vector<vector<size_t>> neighbors, neighborstemp;
+  vector<vector<double>> distances, distancestemp;
+  if (!data::Load("iris.csv", inputdata))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+  if (!data::Load("iris_test.csv", querydata))
+    BOOST_FAIL("Unable to load dataset iris_test.csv!");
+
+  SetInputParam("min", minv);
+  SetInputParam("max", maxv);
+  SetInputParam("distances_file", distancefile);
+  SetInputParam("neighbors_file", neighborsfile);
+  SetInputParam("reference", inputdata);
+
+  mlpackMain();
+
+  RSModel* outputmodel = move(CLI::GetParam<RSModel*>("output_model"));
+
+  SetInputParam("min", minv);
+  SetInputParam("max", maxv);
+  SetInputParam("distances_file", distancefile);
+  SetInputParam("neighbors_file", neighborsfile);
```

Comment:
It would be nice to check that the resulting distances and neighbors are the same also.